### PR TITLE
multi-cluster-diagnostics pod

### PR DESCRIPTION
**Pull request by JesseGoodier**
## What does this PR change?
Creates a new diagnostic deployment for monitoring health in multi-cluster Kubecost environments

## Does this PR rely on any other PRs?
KCM PR: https://github.com/kubecost/kubecost-cost-model/pull/1922

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

Creates a new "diagnostics" deployment that has two functions:
1. On non-primary clusters: the deployment sends Kubecost agent health to a central object-store
2. On the Primary Kubecost cluster: the diagnostics deployment monitors the data sent by the agents and makes this information available via and API
More information available on our docs (link TBD)

## Links to Issues or tickets this PR addresses or fixes
NA

## What risks are associated with merging this PR? What is required to fully test this PR?
Minimal risk. we are using this internally for a month now.

## How was this PR tested?
Many QA deployments

## Have you made an update to documentation? If so, please provide the corresponding PR.
We will have a doc by GA.

### CODE REVIEW

**Review by chipzoller — DISMISSED** [review]
Tests failing as of these changes because `kubecostProductConfigs` is currently in a commented-out state. If diagnostics must be enabled by default, then `kubecostProductConfigs.clusterName` must be enabled by default.

if you see this:
https://github.com/kubecost/cost-analyzer-helm-chart/pull/2780/files#diff-f6d5a9ddec47ad0c6043e1a7258057f261cc6a135446693326d5cd4d27fc1497R124

diagnostics are only enabled if federation is enabled. if federation is enabled, you must set a cluster name.

**Comment by brstuder** [comment]
Let me know what you help you need on this re: documentation

**Comment by nikovacevic** [comment]
Should we change this PR to a draft until the dependencies are merged?

**Comment by chipzoller** [comment]
> Should we change this PR to a draft until the dependencies are merged?

Yes

**Comment by thomasvn** [comment]
@jessegoodier @chipzoller I've cleaned this up a bit, and believe it's ready for review! Code for the diagnostics backend has been merged.

**Comment by jessegoodier** [comment]
@thomasvn this is good by me. mind approving?

**Review by thomasvn — APPROVED** [review]

**Comment by chipzoller** [comment]
Forgot to label this.

File: cost-analyzer/values.yaml [hunk 1]
```diff
@@ -1078,6 +1078,32 @@ kubecostAggregator:
...
   securityContext: {}  # Define a securityContext for the aggregator pod. This will take highest precedence.
 
+## Kubecost Diagnostic Pod (beta, disabled by default until GA) Enables the
+## Kubecost primary to report health/diagnostics of all agent clusters. Each
+## agent cluster sends its health/diagnostic data to a storage bucket. Future
+## versions may include repairing & alerting from the primary.
+## Ref: https://docs.kubecost.com/install-and-configure/install/diagnostics
+##
+diagnostics:
+  enabled: true
```
Comment (chipzoller) [inline cost-analyzer/values.yaml:1088]: This doesn't look disabled by default.

Comment (jessegoodier) [reply cost-analyzer/values.yaml:1088]: We had some conversations with product and engineering and the thought was launch and learn.

I do feel that the risk is low and the potential benefit is high.

Comment (chipzoller) [reply cost-analyzer/values.yaml:1088]: Can I suggest updating the values comment then so it's not confusing?

**Pull request merged**